packages/features/calendar-cache-sql/GoogleCalendarWebhookService.ts

```diff
@@ -0,0 +1,131 @@
+import type { IFeaturesRepository } from "@calcom/features/flags/features.repository.interface";
+import type { CredentialForCalendarService } from "@calcom/types/Credential";
+
+import type { CalendarCacheSqlService } from "./CalendarCacheSqlService";
+import type { ICalendarEventRepository } from "./CalendarEventRepository.interface";
+import type { ICalendarSubscriptionRepository } from "./CalendarSubscriptionRepository.interface";
+
+export interface WebhookRequest {
+  headers: {
+    "x-goog-channel-token"?: string;
+    "x-goog-channel-id"?: string;
+  };
+}
+
+export interface WebhookResponse {
+  status: number;
+  body: Record<string, any>;
+}
+
+export interface WebhookServiceDependencies {
+  subscriptionRepo: ICalendarSubscriptionRepository;
+  eventRepo: ICalendarEventRepository;
+  featuresRepo: IFeaturesRepository;
+  calendarCacheService: CalendarCacheSqlService;
+  getCredentialForCalendarCache: (params: {
+    credentialId: number;
+  }) => Promise<CredentialForCalendarService | null>;
+  webhookToken: string;
+  logger: {
+    debug: (message: string, data?: any) => void;
+    info: (message: string, data?: any) => void;
+    error: (message: string, data?: any) => void;
+  };
+}
+
+export class GoogleCalendarWebhookService {
+  constructor(private dependencies: WebhookServiceDependencies) {}
+
+  async processWebhook(request: WebhookRequest): Promise<WebhookResponse> {
+    const { headers } = request;
+    const { logger, webhookToken } = this.dependencies;
+
+    const channelToken = headers["x-goog-channel-token"];
+    const channelId = headers["x-goog-channel-id"];
+
+    logger.debug("Processing webhook", { channelToken, channelId });
+
+    // Validate webhook token
+    if (channelToken !== webhookToken) {
+      return {
+        status: 403,
+        body: { message: "Invalid API key" },
+      };
+    }
+
+    // Validate channel ID
+    if (typeof channelId !== "string") {
+      return {
+        status: 403,
+        body: { message: "Missing Channel ID" },
+      };
+    }
+
+    try {
+      // Check if SQL cache write is enabled globally
+      const isSqlWriteEnabled = await this.dependencies.featuresRepo.checkIfFeatureIsEnabledGlobally(
+        "calendar-cache-sql-write"
+      );
+
+      if (!isSqlWriteEnabled) {
+        logger.debug("SQL cache write not enabled globally");
+        return {
+          status: 200,
+          body: { message: "ok" },
+        };
+      }
```

Comment:
DevinAI, this violates separation of concerns. The service shouldn not care of feature flags. This should happen upstream in the webhook code.